AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet4Solutions.lean

```diff
@@ -0,0 +1,200 @@
+/-
+This is a solutions sheet.
+-/
+
+import Mathlib.Tactic
+
+/-Credit for some exercises goes to Kevin Buzzard and his Formalizing Mathematics
+course OR to the Natural Number Game. If you would like to learn more about Lean,
+Buzzard's course goes more in depth in relation to numerous undergraduate math topics.
+When exercises are from either of these sources, they will be clearly marked so as
+to give credit.-/
+
+/-The tactic constructor should be completely new to you. It is very useful for
+deconstructing goals that use "and" (∧). It is similar to the cases tactic in the
+way that it breaks up one goal into multiple goals to be completed, but it is
+used on the goal rather than on a hypothesis.
+
+Let's take a look at the example below:-/
+
+variable (P Q R S T : Prop)
+
+/-FM Section 1 Sheet 4-/
+example : P → Q → P ∧ Q := by
+  intro h1 h2
+  constructor
+  · exact h1
+  · exact h2
+  done
+
+/-Note that constructor breaks the goal of P ∧ Q into two goals, ⊢ P and then ⊢ Q.
+
+With the constructor tactic comes the need to talk about some helpful formatting
+in Lean. The · that you see above is used to indicate steps to solve separate goals.
+⊢ P is one goal, so a new dot is added. exact h1 then solves that goal.
+⊢ Q is the next goal, so another dot is added. exact h2 then solves that goal.
+In the case where more than one tactic is requred to resolve the goal, lines following
+the line with the new · are indented to indicate that they are associated with a
+certain goal.
+
+For example:-/
+
+/-FM Section 1 Sheet 5-/
+example : (P ↔ Q) ↔ (Q ↔ P) := by
+  constructor
+  · intro h1
+    rw[h1]
```

Comment:
it is standard to put a space between rw and [

Reply:
Fixed and also fixed all other instances of this with rw.